app/api/users.py

```diff
@@ -108,9 +108,12 @@ class UserDetail(ResourceDetail):
     """
 
     def before_get(self, args, kwargs):
-        if current_user.is_admin or current_user.is_super_admin or current_user:
+        if current_user:
             self.schema = UserSchema
-        else:
+        elif 'Authorization' in request.headers and (has_access('is_admin') or has_access('is_super_admin')):
```

Comment:
or can be used, it's the same branch

Reply:
ok, instead I should remove authorization check or everyone will be able to access the complete use profile

Reply:
No, everyone should not be able to check complete user profile

Reply:
Yess